src/net_processing.cpp

```diff
@@ -1214,6 +1226,40 @@ void PeerLogicValidation::NewPoWValidBlock(const CBlockIndex *pindex, const std:
     });
 }
 
+static bool UpdateCFHeadersCache(const BlockFilterIndex& filter_index)
+{
+    LOCK(cs_main);
+    std::lock_guard<std::mutex> _lock(active_chain_cf_headers_mtx);
+
+    const CChain& active_chain = ::ChainActive();
+
+    // Drop any entries in the checkpoint cache that have been reorganized off the active chain.
+    int new_size = active_chain_cf_headers.size();
```

Comment:
Suggestion: consider simplifying this bit to remove these manual "off-by-one'ness-ness" like `>`, `-1`, etc, using reverse iterators and `find_if`, with something like this:
```
const auto it = std::find_if(active_chain_cf_headers.crbegin(), active_chain_cf_headers.crend(), [&](const std::pair<const CBlockIndex*, uint256> &i) {
    return active_chain.Contains(i.first);
});
active_chain_cf_headers.erase(it.base(), active_chain_cf_headers.cend());
```
later `(new_size + 1)` obviously can be changed into `active_chain_cf_headers.size()`.

This method be toyed-with using this trivial example:
```
#include <algorithm>
#include <iostream>
#include <iterator>
#include <vector>

int main()
{
    std::vector<int> vals{1, 1, 0, 0, 0};

    const auto it = std::find_if(vals.crbegin(), vals.crend(), [](int v) -> bool {
        return v > 0;
    });

    vals.erase(it.base(), vals.cend());

    std::copy(vals.cbegin(),
              vals.cend(),
              std::ostream_iterator<int>(std::cout, " "));
}
```